File: fastpackandroid/app/src/main/java/com/fastpack/fastpackandroid/activity/MainActivity.java
package com.fastpack.fastpackandroid.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.fastpack.fastpackandroid.R;
import com.fastpack.fastpackandroid.interfaces.Interfaces;
import com.fastpack.fastpackandroid.layout.LayoutMainActivity;

public class MainActivity extends ActivityBasic {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setActionBar();
    }

    @Override
    public int getIdLayout() {
        return R.layout.activity_main;
    }

    @Override
    public Interfaces.LayoutBasicMethods getLayoutBasic() {
        return new LayoutMainActivity( this );
    }

    @Override
    public int getIdContainerView() {
        return R.id.container;
    }

}

File: fastpackandroid/app/src/main/java/com/fastpack/fastpackandroid/activity/MainActivity.kt
package com.fastpack.fastpackandroid.activity

import android.content.Intent
import android.os.Bundle

import com.fastpack.fastpackandroid.R
import com.fastpack.fastpackandroid.interfaces.Interfaces
import com.fastpack.fastpackandroid.layout.LayoutMainActivity

class MainActivity : ActivityBasic() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        startActivity( Intent(this,ActivityPedidoCriar::class.java) )
        setActionBar()
    }

    override fun getIdLayout(): Int {
        return R.layout.activity_main
    }

    override fun getNewInstanceOfLayoutBasic(): Interfaces.LayoutMethodsRequierieds {
        return LayoutMainActivity(this)
    }

    override fun getIdContainerView(): Int {
        return R.id.container
    }

}
